From: Changbin Du <changbin.du@huawei.com>
Subject: modules-wait-do_free_init-correctly-fix
Date: Mon, 25 Dec 2023 14:15:13 +0800

fix compilation issue for no CONFIG_MODULES found by 0-DAY

Link: https://lkml.kernel.org/r/20231225061513.2984575-1-changbin.du@huawei.com
Fixes: 1a7b7d922081 ("modules: Use vmalloc special flag")
Signed-off-by: Changbin Du <changbin.du@huawei.com>
Cc: Xiaoyi Su <suxiaoyi@huawei.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 include/linux/moduleloader.h |    6 ++++++
 1 file changed, 6 insertions(+)

--- a/include/linux/moduleloader.h~modules-wait-do_free_init-correctly-fix
+++ a/include/linux/moduleloader.h
@@ -115,7 +115,13 @@ int module_finalize(const Elf_Ehdr *hdr,
 		    const Elf_Shdr *sechdrs,
 		    struct module *mod);
 
+#ifdef CONFIG_MODULES
 void flush_module_init_free_work(void);
+#else
+static inline void flush_module_init_free_work(void)
+{
+}
+#endif
 
 /* Any cleanup needed when module leaves. */
 void module_arch_cleanup(struct module *mod);
_
